### PR TITLE
Add support for the Internal Ranges API

### PR DESCRIPTION
**Pull request by daanheikens**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/14698

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_network_connectivity_internal_range
```

### CODE REVIEW

File: mmv1/products/networkconnectivity/InternalRanges.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      User-defined labels.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+  - !ruby/object:Api::Type::String
+    name: 'ipCidrRange'
```
Comment (rileykarson) [inline mmv1/products/networkconnectivity/InternalRanges.yaml:94]: This needs `default_from_api: true` to cover the automatic assignment case!

File: mmv1/products/networkconnectivity/InternalRanges.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'InternalRange'
+base_url: 'projects/{{project}}/locations/global/internalRanges'
+create_url: 'projects/{{project}}/locations/global/internalRanges?internalRangeId={{name}}'
+self_link: 'projects/{{project}}/locations/global/internalRanges/{{name}}'
+update_verb: :PATCH
```
Comment (rileykarson) [inline mmv1/products/networkconnectivity/InternalRanges.yaml:19]: We'll need at least one update test, see https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test for instructions

File: mmv1/products/networkconnectivity/InternalRanges.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    description: |
+      Output only. The list of resources that refer to this internal range.
+      Resources that use the internal range for their range allocation are referred to as users of the range.
+      Other resources mark themselves as users while doing so by creating a reference to this internal range. Having a user, based on this reference, prevents deletion of the internal range referred to. Can be empty.
+  - !ruby/object:Api::Type::Array
+    name: 'overlaps'
+    description: |
+      Optional. Types of resources that are allowed to overlap with the current internal range.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'undefined'
```
Comment (rileykarson) [inline mmv1/products/networkconnectivity/InternalRanges.yaml:146]: note to self: This `name` value never gets used for an array of enums, so this is reasonable.

**Review by rileykarson — COMMENTED** [review]
Couple notes on tests, the resource code itself looks good so far!

**Comment by daanheikens** [comment]
> Couple notes on tests, the resource code itself looks good so far!

Thanks @rileykarson i will have a look at it!

File: mmv1/products/networkconnectivity/InternalRanges.yaml (hunk 2 above)
Comment (daanheikens) [reply mmv1/products/networkconnectivity/InternalRanges.yaml:19]: @rileykarson Added an update test, happy to receive your feedback!

File: mmv1/products/networkconnectivity/InternalRanges.yaml (hunk 1 above)
Comment (daanheikens) [reply mmv1/products/networkconnectivity/InternalRanges.yaml:94]: Done!

File: mmv1/products/networkconnectivity/InternalRanges.yaml (hunk 2 above)
Comment (daanheikens) [reply mmv1/products/networkconnectivity/InternalRanges.yaml:19]: Something pops to mind, i might want to add another update test specifically for the `ipCidrRange` field to combine it with the `usage` field. I believe these are now not covered. How may i add another update test? I assume it's replicating the process?

Comment (rileykarson) [reply mmv1/products/networkconnectivity/InternalRanges.yaml:19]: Yep, another TestAcc block and associated configs.

**Comment by rileykarson** [comment]
(I think that message is wrong, I'll let you know if it's not during my next full pass)

File: mmv1/products/networkconnectivity/InternalRanges.yaml (hunk 2 above)
Comment (daanheikens) [reply mmv1/products/networkconnectivity/InternalRanges.yaml:19]: @rileykarson Added another test, it should cover now all updatable fields.

**Review by rileykarson — APPROVED** [review]
LGTM, thank you!

**Pull request merged**